### PR TITLE
Add option to tunnel client to specify a root certificate authority

### PR DESCRIPTION
**Pull request by goriccardo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,10 +101,11 @@ looks like this
 Configuration options:
 
 * `server_addr`: server TCP address, i.e. `54.12.12.45:4443`
-* `insecure_skip_verify`: controls whether a client verifies the server's certificate chain and host name, if using self signed certificates must be set to `true`, *default:* `false`
+* `insecure_skip_verify`: controls whether a client should skip the verification of the server's certificate chain and host name. If set to `true` the client will accept *any* server certificate as valid, *default:* `false`
```
Comment (mmatczuk) [inline README.md]: :+1:

File: cmd/tunnel/config.go [hunk 2]
```diff
@@ -44,6 +44,7 @@ type Tunnel struct {
 type ClientConfig struct {
 	ServerAddr         string             `yaml:"server_addr"`
 	InsecureSkipVerify bool               `yaml:"insecure_skip_verify"`
+	RootCA             string             `yaml:"root_ca"`
```
Comment (mmatczuk) [inline cmd/tunnel/config.go]: Could we move it down to keep the order consistent.

File: cmd/tunnel/config.go [hunk 3]
```diff
@@ -59,6 +60,7 @@ func loadClientConfigFromFile(file string) (*ClientConfig, error) {
 	c := ClientConfig{
 		TLSCrt: filepath.Join(filepath.Dir(file), "client.crt"),
 		TLSKey: filepath.Join(filepath.Dir(file), "client.key"),
+		RootCA: "",
```
Comment (mmatczuk) [inline cmd/tunnel/config.go]: No need to do that.

File: cmd/tunnel/tunnel.go [hunk 4]
```diff
@@ -106,10 +120,16 @@ func main() {
 	}
 }
 
-func tlsConfig(cert tls.Certificate, c *ClientConfig) *tls.Config {
+func tlsConfig(cert tls.Certificate, rootca *x509.CertPool, c *ClientConfig) *tls.Config {
```
Comment (mmatczuk) [inline cmd/tunnel/tunnel.go]: Could you change the signature of this function to `func tlsConfig(config *ClientConfig) (*tls.Config, error)` and move all the tls related processing there.

Comment (mmatczuk) [reply cmd/tunnel/tunnel.go]: Also it might be good to rename `c` to `config` in the main function.

File: cmd/tunnel/tunnel.go [hunk 5]
```diff
@@ -86,6 +88,18 @@ func main() {
 		fatal("failed to load certificate: %s", err)
 	}
 
+	var rootca *x509.CertPool
+	if c.RootCA != "" {
+		rootca = x509.NewCertPool()
```
Comment (mmatczuk) [inline cmd/tunnel/tunnel.go]: This is usually called `roots`.

File: cmd/tunnel/tunnel.go [hunk 6]
```diff
@@ -86,6 +88,18 @@ func main() {
...
+		capem, err := ioutil.ReadFile(c.RootCA)
```
Comment (mmatczuk) [inline cmd/tunnel/tunnel.go]: How about `rootPEM`?

**Comment by goriccardo** [comment]
@mmatczuk, thank you for the review, I applied all your suggestions.

**Comment by mmatczuk** [comment]
Thanks!

**Pull request merged**